src/storage/index.cpp

```diff
@@ -49,19 +49,31 @@ void Index::Delete(DataChunk &entries, Vector &row_identifiers) {
 }
 
 bool Index::MergeIndexes(Index &other_index) {
+
 	IndexLock state;
 	InitializeLock(state);
 
 	switch (this->type) {
-	case IndexType::ART: {
-		auto &art = Cast<ART>();
-		return art.MergeIndexes(state, other_index);
-	}
+	case IndexType::ART:
```

Comment:
Unrelated to this PR - but the problem with these types of switches is that it does not work for potential indexes defined by extensions. At some point these will need to be converted to virtual methods or similar instead.

Reply:
That makes sense, should I do this now? I can also add it to the next PR.

Reply:
Next PR is fine